pkg/apis/clusteroperator/types.go

```diff
@@ -151,6 +159,11 @@ type ClusterStatus struct {
 	// For machine set hardware, see the status of each machine set resource.
 	Provisioned bool
 
+	// ProvisioningJob is the reference to the Job performing infrastructure provisioning
+	// for this cluster. It will be set while the infrastructure is being provisioned.
+	// This reference is guaranteed to be nil if Provisioned is true
```

Comment:
We're not saying that the job will go away, simply that we won't keep a reference to it in the cluster status when not provisioning the cluster. I added the comment because @smarterclayton asked for it.

Reply:
Sure, the job won't go away. But without the reference, it may not be easy for an end user to find the job.